lib/form/ml_form.dart

```diff
@@ -0,0 +1,136 @@
+import 'package:flutter/material.dart';
+import 'package:memory_lamp/helpers/size_mq.dart';
+import 'package:memory_lamp/theming/ml_colors.dart';
+
+class MLForm extends StatefulWidget {
+  final bool loadForSignup;
+  const MLForm({this.loadForSignup = false});
+  @override
+  _MLFormState createState() => _MLFormState();
+}
+
+class _MLFormState extends State<MLForm> {
+  final _mlFormKey = GlobalKey<FormState>();
+
+  @override
+  Widget build(BuildContext context) {
+    return Form(
+      key: _mlFormKey,
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.spaceAround,
+        mainAxisSize: MainAxisSize.max,
+        children: <Widget>[
+          _userNameField(),
+          SizedBox(height: SizeMQ.height! * .03),
+          if (widget.loadForSignup) _emailField(),
+          SizedBox(height: SizeMQ.height! * .03),
+          _passwordField(),
+          SizedBox(height: SizeMQ.height! * .03),
+          if (widget.loadForSignup) _confirmPasswordField(),
+          widget.loadForSignup ? _agreement() : _rememberMe(),
+        ],
+      ),
+    );
+  }
+}
+
+//username
+TextFormField _userNameField() => TextFormField(
+      keyboardType: TextInputType.name,
+      decoration: _defaultInputDecoration(label: 'Username'),
+    );
+
+//email
+TextFormField _emailField() => TextFormField(
+      keyboardType: TextInputType.emailAddress,
+      decoration: _defaultInputDecoration(label: 'Email'),
+    );
+
+//password
+TextFormField _passwordField() => TextFormField(
+      keyboardType: TextInputType.visiblePassword,
+      obscureText: true,
+      decoration: _defaultInputDecoration(label: 'Password'),
+    );
+
+//confirmPassword
+TextFormField _confirmPasswordField() => TextFormField(
+      keyboardType: TextInputType.visiblePassword,
+      obscureText: true,
+      decoration: _defaultInputDecoration(label: 'Confirm Password'),
+    );
+
+//AgreeToTos
+// ---> i get errors when i put this above. can't seem find the error. for the meantime i put it here
```

Comment:
I think alam ko kung bakit may error to, 
di ko pa lang natetest pero most likely dahil labas na ng scope yung widgets mo,

yung bracket sa line 35 try mo ilipat sa pinaka dulo ng file (line 135)
![35](https://user-images.githubusercontent.com/62323308/130418336-506a8458-1b03-4164-8d4b-2f2a2fcbf6a4.png)
![35](https://user-images.githubusercontent.com/62323308/130418476-ae514d74-a412-4968-8d4b-f0e33c47c7b1.png)

 yung pinaka baba yung may TO DO: validations,

Reply:
oowww yun yung problema ko undefined variables, noted salamat po